FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java

```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
     public void findMatch() {
         SharedPreferences preferences = getSharedPreferences("PREFS", 0);
         String[] fileNameList = preferences.getString("fileNameList", "").split(" ");
+        ArrayList<Pair<String, Integer>> matches = new ArrayList<>();
 
-        double maxRatio = 0;
-        String bestFileName = null;
+        //double maxRatio = 0;
+        //String bestFileName = null;
         for (String fileName : fileNameList) {
             double ratio = matchFeaturesFile(fileName);
-            if (ratio > maxRatio) {
-                maxRatio = ratio;
-                bestFileName = fileName;
-            }
+//            if (ratio > maxRatio) {
+//                maxRatio = ratio;
+//                bestFileName = fileName;
+//            }
+            matches.add(new Pair<>(fileName,(int) (ratio * 100)));
         }
 
         AlertDialog.Builder builder = new AlertDialog.Builder(this);
         builder.setTitle("Match found");
```

Comment:
Before I added the listview it would also always show match found. Can change to match not found if not files present.